### ISSUE TITLE
Add fetch config into layer spec.

### ISSUE THREAD

**Issue by sorodrigo** [Vizzuality/layer-manager#78]
The Layer Manager is too opinionated about when should a layer unmount. A layer will update if the config changes, and only update if the interactivity params have changed (opacity, visibility, decodeParams, zIndex).

This prevents us from animating layers using mapbox style expressions and probably other use cases out there.

I propose:

- Separating the fetching from the client side definition of the layers:
```js
{
  id: 1234,
  name: "my layer",
  map: {
    maxzoom: 3,
    minzoom: 2,
    vectorLayers: [
      {
        filter: ["==", "year", year],
        paint: {
          "fill-color": ["interpolate", ["linear"], ["get", "probabilityemca"]],
          "fill-opacity": opacity
        },
        "source-layer": "layer0",
        type: "fill"
      }
    ]
  },
  fetch: {
    account: "my-account",
    body: {
      layers: [
        {
          options: {
            sql:
              "SELECT cartodb_id, the_geom_webmercator, uuid, iso3 FROM all_geometry {{where}}) .."
          },
          type: "cartodb"
        }
      ]
    },
    params_config: [],
    sql_config: [
      {
        key: "where",
        key_params: [
          {
            key: "iso3",
            required: true
          }
        ]
      }
    ],
    sqlParams: {
      where: {
        iso3: iso
      }
    },
  }
};
```
This allows us to decouple providers from presentation and also, makes the updating of the layer more coherent.

**Comment by mbarrenechea** [comment]
I like your approach but I will change some names:

```json
{
  id: 1234,
  name: "my layer",
  render: {
    maxzoom: 3,
    minzoom: 2,
    styles: [
      {
        filter: ["==", "year", year],
        paint: {
          "fill-color": ["interpolate", ["linear"], ["get", "probabilityemca"]],
          "fill-opacity": opacity
        },
        "source-layer": "layer0",
        type: "fill"
      }
    ]
  },
  source: {
    account: "my-account",
    layers: [
      {
        options: {
          sql: "SELECT cartodb_id, the_geom_webmercator, uuid, iso3 FROM all_geometry ..."
        },
        type: "cartodb"
      }
    ]
  },
  config: {
    params_config: [
      { key: 'year', required: true, default: 2001 }
    ],
    sql_config: []
  },
  params: {
    year: 2001
  },
  sqlParams: {},
  decodeParams: {}
};
```

**Comment by mbarrenechea** [comment]
Solved in #80
